Updated text of Introduction to WCAG (#313)
wpaccessibility/wp-a11y-docs: docs/start/wcag-intro/index.md

parent: Start with accessibility
description: What are the Web Content Accessibility Guidelines and how to use them.
nav_order: 3
contributors:
  - Annelies Verhelst
  - Rian Rietveld
---

# Introduction to WCAG

> When designing, writing, developing or testing a web page, plugin, theme, app or other software, you will be commonly referred to the Web Content Accessibility Guidelines, or WCAG in short. Following these guidelines will make it accessible for as many people as possible, including those with disabilities. 

Reviewer: I think we should lead with what WCAG is, rather than with when you might need them. I'd also like to see this in an active voice, instead of passive. Something like:

"The Web Content Accessibility Guidelines, or WCAG, are the primary guidelines for meeting accessibility standards worldwide. You will need to refer to them when designing, writing, developing, or testing web pages, plugins, themes, or other software."

> The guidelines are globally used in legislation and normative documents, and are created and maintained by the Web Accessibility Initiative (WAI), a part of the international World Wide Web Consortium (W3C).

Reviewer: I'd strike the phrase "and normative documents" here. The guidelines *are* normative documents, so I'm not sure that really makes sense.

> It's worth mentioning that WCAG in itself is not a law. They are guidelines for proper web development. Be aware that it's very often referred to as something that's legally mandatory. 

Reviewer: ```suggestion
It's worth mentioning that WCAG in itself is not a law. It is a set of guidelines for proper web development. Be aware that it's very often referred to as something that's legally mandatory. 
```

> However, there's more nuance to that. Correctly following these guidelines and techniques will likely help ensure compliance with accessibility laws and legislation. [More about legislation and laws for accessibility](https://wpaccessibility.org/docs/topics/legislation/).

Reviewer: ```suggestion
However, there's more nuance to that. Correctly following these guidelines and techniques will help ensure compliance with accessibility laws and legislation. [More about legislation and laws for accessibility](https://wpaccessibility.org/docs/topics/legislation/).
```

parent: Start with accessibility
description: What are the Web Content Accessibility Guidelines and how to use them.
nav_order: 3
contributors:
  - Annelies Verhelst
  - Rian Rietveld
---

# Introduction to WCAG

When designing, writing, developing or testing a web page, plugin, theme, app or other software, you will be commonly referred to the Web Content Accessibility Guidelines, or WCAG in short. Following these guidelines will make it accessible for as many people as possible, including those with disabilities. 

The guidelines are globally used in legislation and normative documents, and are created and maintained by the Web Accessibility Initiative (WAI), a part of the international World Wide Web Consortium (W3C).

> It's worth mentioning that WCAG in itself is not a law. They are guidelines for proper web development, as there are guidelines to use HTML, PHP and other programming languages as well. Be aware that it's very often referred to as something that's legally mandatory. 

Author: For:
They are guidelines for proper web development, as there are guidelines to use HTML, PHP and other programming languages as well.

Propose:
They are guidelines for proper frontend development

However, there's more nuance to that. Correctly following these guidelines and techniques will likely help ensure compliance with accessibility laws and legislation. [More about legislation and laws for accessibility](https://wpaccessibility.org/docs/topics/legislation/).

## WCAG Versions

At this point (early 2026) in most countries that have legislation about web accessibility, [WCAG version 2.1](https://www.w3.org/TR/WCAG21/) is the version being referred to when they speak of web accessibility standards. It contains 78 criteria, divided into 13 general guidelines. 

Many countries are starting to use [version 2.2](https://www.w3.org/TR/WCAG22/) as the version of choice. The main difference is that this one has slightly more criteria (86) and a few were removed or changed levels. In this documentation, we use version 2.2 as our point of reference.

> The current versions are mainly focused on complete websites and apps. There is a newer version (WCAG 3.0) in development that should provide an outcomes-based conformance model, aiming to better cover cognitive and low-vision disabilities. It will also cover a broader scope, including mobile apps, AR/VR software, and apps for other devices, such as TVs

Author: Add a . at the end

> such as TVs.

> As of version WCAG 3.0, the abbreviation will reflect the wide scale it can be used for. It will then be named the W3C Accessibility Guidelines. [More on version 3.0(https://www.w3.org/TR/wcag-3.0-explainer/).

Author: `[More on version 3.0](https://www.w3.org/TR/wcag-3.0-explainer/).`

## WCAG success criteria

> The success criteria can be seen as a set of rules to follow. Some of them are very specific; others are more open to interpretation, because otherwise there would be no room for creativity. 

Author: The success criteria can be seen as a set of rules on how to comply to a guideline.

Author: > Some of them are very specific; others are more open to interpretation, because otherwise there would be no room for creativity. 

I would leave that out. We as auditors know this, but it may be confusing to readers.

> They describe a general focus, like "Make sure text has sufficient contrast", exceptions, specifics, and/or use cases. The criteria are numbered, for example, 2.1.1 Keyboard, meaning it's part of Chapter/Principle 2 and guideline 2.1 (Keyboard Accessible).

Author: Break this op in multiple sentences and simplify.

Each success criterion has a number, a name and a level.

The number is composed of:
- the number of the principle
- the number of the guideline
- the number of the criterion itself
 
For example the success criterion **2.1.1 Keyboard, Level A** means it's part of principle 2 (Operable), guideline 2.1 (Keyboard Accessible) and the first criterion for that guideline.

> ## WCAG Levels of WCAG

Reviewer 2: minus one WCAG (WCAG levels of WCAG)

- Can visitors comprehend all content? (Understandable)
- Can visitors use any device? (Robust)

> These are the four principles in WCAG, and they reflect what a website should be: perceivable, operable, understandable and robust.

Reviewer: ```suggestion
There are four key principles in WCAG, and they reflect what a website should be: **perceivable**, **operable**, **understandable** and **robust**.
```

- Content must be able to be interpreted reliably by a wide variety of user agents, including assistive technologies.
- All user interface components should have programmatically determinable names and roles.

> ### Non-interference section 5.2.5

Reviewer: It would probably be a good idea to add a brief summary of section 5 before going into this subsection; otherwise, it's not clear what the context of this is. Something like:

"Sections 1 through 4 contain the specific guidelines used to create accessible content. Section 5, however, talks about what it means to conform to these guidelines and defines important terminology used for conformance."

> Although when following guidelines you will often stay within chapters 1 to 4 (the four principles) of WCAG, it's necessary to read through [section 5.2.5](https://www.w3.org/TR/WCAG22/#cc5) as well. It's about non-interference, meaning your techniques cannot block users' ability to access any part of a page. 

Reviewer: Use the term "section" instead of "chapter". This may need some rewriting after adding a general paragraph about section 5.

- 1.4.2: Audio control
- 2.1.1: No keyboard trap
- 2.3.1: Three flashes or below threshold
- 2.2.2: Pause, stop, hide

Not passing these success criteria can make it hard, if not impossible, for some users to continue browsing or using your app. It will interfere with their physical or mental health or assistive devices.

### WCAG success criteria

The success criteria can be seen as a set of rules for complying with a guideline. Some of them are very specific; others are more open to interpretation. 

Each success criterion has a number, a name and a level.

The number is composed of:

- the number of the principle
- the number of the guideline
- the number of the criterion itself

For example, the success criterion 2.1.1 Keyboard, Level A means it's part of principle 2 (Operable), guideline 2.1 (Keyboard Accessible) and the first criterion for that guideline.

### WCAG Levels of accessibility

There are 3 levels of accessibility:

- A (basic)
- AA (the global accessibility standard)
- AAA (for dedicated assistive support)

> The success criteria can have one of 3 levels, A, AA or AAA. In general, level A criteria consist of basic rules that should lead to a technically functional website or app. Nothing too special. Level AA consist of criteria that, if you follow them correctly, will lead to a website or app that works for most known assistive devices. 

Reviewer: ```suggestion
The success criteria can have one of 3 levels, A, AA or AAA. In general, level A criteria consist of basic rules that should lead to a technically functional website or app. Level AA consist of criteria that, if you follow them correctly, will lead to a website or app that works for most known assistive devices. 
```

> Passing tests for the success criteria of level A and AA is what most legislation asks of you, and is what is meant when people say you are WCAG compliant. Level AAA adds additional success criteria that are necessary for most disabled people to use your website or app. 

Reviewer: ```suggestion
Passing tests for the success criteria of level A and AA is what most legislation asks of you, and is what is meant when people say you are WCAG compliant. Level AAA adds additional success criteria that are necessary to expand the number of people with disabilities who can use your website or app. 
```

> They're not necessarily harder to implement, but they're useful for a smaller group and sometimes require more work to maintain. Also, it is implied that when you say you (need to) comply with level AA, you automatically comply with A as well.

Reviewer: The second sentence here is out of place; you're talking about level AAA, so suddenly diverting to address AA and A doesn't make sense. This should probably be referenced earlier, when talking about level AA. 

Rephrase, move earlier:

"It is implied that when you need to comply with level AA, you automatically need to comply with level A as well."

### WCAG Versions

> At this point (early 2026) in most countries that have legislation about web accessibility, [WCAG version 2.1](https://www.w3.org/TR/WCAG21/) is the version being referred to when they speak of web accessibility standards. It contains 78 criteria, divided into 13 general guidelines. 

Reviewer: ```suggestion
As of early 2026, in most countries that have legislation about web accessibility, [WCAG version 2.1](https://www.w3.org/TR/WCAG21/) is the version referenced. It contains 78 criteria, divided into 13 general guidelines. 
```

> Many countries are starting to use [version 2.2](https://www.w3.org/TR/WCAG22/) as the version of choice. The main difference is that this one has slightly more criteria (86) and a few were removed or changed levels. In this documentation, we use version 2.2 as our point of reference.

Reviewer: ```suggestion
Many countries are starting to use [version 2.2](https://www.w3.org/TR/WCAG22/) as the version of choice. WCAG 2.2 has added a few more criteria (86 in total) and a couple criteria were removed or changed levels. In this documentation, we use version 2.2 as our point of reference.
```

> The current versions are mainly focused on complete websites and apps. There is a newer version (WCAG 3.0) in development that should provide an outcomes-based conformance model, aiming to better cover cognitive and low-vision disabilities. It will also cover a broader scope, including mobile apps, AR/VR software, and apps for other devices, such as TVs.

Reviewer: ```suggestion
The current versions are mainly focused on complete websites and apps. There is a newer version (WCAG 3.0) in development that intends to provide an outcome-based conformance model, aiming to better cover cognitive and low-vision disabilities. It will also cover a broader scope, including mobile apps, AR/VR software, and apps for other devices, such as TVs.
```

As of version WCAG 3.0, the abbreviation will reflect the wide scale it can be used for. It will then be named the W3C Accessibility Guidelines. [More on version 3.0](https://www.w3.org/TR/wcag-3.0-explainer/).

## Implementing the guidelines

> The guidelines and criteria do not provide specific techniques for passing them in testing. However, the W3C provided a [quick reference guide](https://www.w3.org/WAI/WCAG22/quickref/) with common techniques and examples for implementing them. When designing, writing or coding, keep asking yourself if what you're doing contributes to the main purpose of the guideline. 

Reviewer: ```suggestion
The guidelines and criteria do not provide specific techniques to achieve conformance. However, the W3C provided a [quick reference guide](https://www.w3.org/WAI/WCAG22/quickref/) with common techniques and examples for implementing them. When designing, writing or coding, keep asking yourself if what you're doing contributes to the main purpose of the guideline. 
```

> It's easy to get caught in techniques and specifics, while the purpose is to follow a guideline and be able to check off a success criterion. 

Reviewer: I think this needs to be rewritten; it makes it sound like we're encouraging a checkbox mentality.

Author: I'll remove it, it's already stated in the previous sentence.

Example:

> - Guideline: You need to make sure all texts have enough contrast with the background.

Reviewer: ```suggestion
- Guideline: You need to make sure all text has enough contrast with the background.
```

Author: I added the text of the actual guideline:
Make it easier for users to see and hear content including separating foreground from background.

> - Success criterion: It must be at least 4.5:1 if text is smaller than 14 pt (and some other specifics).

Reviewer: Instead of a vague "and some other specifics", can this add a link to the color contrast page? e.g. https://wpaccessibility.org/docs/topics/design/color/color-contrast-text/#color-contrast-of-text-against-its-background

I think it's fine for this example to be incomplete; but we should make it as easy as possible to get more detail.

Perhaps we can also just omit the example and link across to the page instead.

Author: removing it

- Technique: Use a color contrast checker and CSS to make sure all texts pass this criterion.

## Testing or getting tested

> You may have seen audit reports in which a website was tested for accessibility. The audits mostly follow the [WCAG Evaluation Method](https://www.w3.org/WAI/test-evaluate/conformance/wcag-em/). This method provides auditors and testers with guidelines for checking whether a success criterion has been met (and thus whether a guideline has been followed correctly). [More about testing](https://wpaccessibility.org/docs/testing}.

Reviewer: I don't think we should say "the audits mostly follow the [eval method]"; that's certainly not true. People may have seen many audits following many different methodologies. We can certainly have a section that talks about this method, but let's not presume that anybody has necessarily seen one like it before.

> Per success criterion, usually many findings can be reported. That's why a report could be overwhelming, especially when you haven't done any work on the website yet. It's recommended to try to implement techniques and follow the guidelines before you get it audited. 

Reviewer: Rephrase for clarity:

"Many finding can be reported for each success criterion in the report. That can make a report overwhelming! It's recommended that you try and implement best practice accessibility techniques and follow the guidelines to the best of your abilities before getting an audit."

This way, it will be easier to follow recommendations from the report. It can be useful to read through the evaluation method yourself, as it offers ways to check for accessibility issues.

## Browser support

Decide which browsers and versions your theme or plugin will support. Some [WCAG techniques](https://www.w3.org/WAI/WCAG22/Understanding/understanding-techniques) apply only to older browsers, e.g, Internet Explorer 7 and below. Don't forget to check mobile browsers and software versions.

> WordPress no longer supports Internet Explorer as of version 5.8.

Reviewer: Instead of specifying this one particular support case, let's link to the WordPress browser support page: https://make.wordpress.org/core/handbook/best-practices/browser-support/.

## Relation to ISO standards, EAA and EN 301 549

> Under [ISO/IEC 40500:2025](https://www.iso.org/standard/91029.html), WCAG 2.2 is an approved standard. This means you can use WCAG free from ISO. 

Reviewer: I don't understand the last sentence... "This means you can use WCAG free from ISO."

> In most laws in Europe that have the [European Accessibility Act](https://commission.europa.eu/strategy-and-policy/policies/justice-and-fundamental-rights/disability/european-accessibility-act-eaa_en) implemented, the European norm for accessibility [EN 301 549](https://accessible-eu-centre.ec.europa.eu/content-corner/digital-library/en-3015492021-accessibility-requirements-ict-products-and-services_en) is mentioned for accessibility requirements. This document explains accessibility requirements for all ICT products and services and uses WCAG 2.1 as its basis. It's expected to be updated to WCAG 2.2 late 2026.

Reviewer: We need to expand the abbreviation 'ICT'; not all users will know that.

- 1.4.2: Audio control
- 2.1.1: No keyboard trap
- 2.3.1: Three flashes or below threshold
- 2.2.2: Pause, stop, hide

Not passing these success criteria can make it hard, if not impossible, for some users to continue browsing or using your app. It will interfere with their physical or mental health or assistive devices.

## Implementing the guidelines

The guidelines and criteria do not provide specific techniques for passing them in testing. However, the W3C provided a [quick reference guide](https://www.w3.org/WAI/WCAG22/quickref/) with common techniques and examples for implementing them. When designing, writing or coding, keep asking yourself if what you're doing contributes to the main purpose of the guideline. 

It's easy to get caught in techniques and specifics, while the purpose is to follow a guideline and be able to check off a success criterion. 

Example:

- Guideline: You need to make sure all texts have enough contrast with the background.
- Success criterion: It must be at least 4.5:1 if text is smaller than 14 pt (and some other specifics).

> - Technique: Use a color contrast checker and CSS to make sure all texts pass this criterion.

Author: Why add the "and CSS"?

Reviewer 2: because you need to change the CSS based on the outcome of the color contrast checker and make sure it gets done right everywhere. Techniques are not limited to the tools alone, you need to implement it too. Read it as "Use CSS to make sure all texts (on the website) pass this criterion".

Reviewer: I think that adding "use CSS" is redundant. CSS is the method used for *all* color changes, so it's fundamentally essential, and doesn't need to be mentioned. And if somebody did use an alternate method (e.g., a font attribute), that doesn't technically violate WCAG, and would be acceptable...although horrific.

Author: removing it

## Testing or getting tested

You may have seen audit reports in which a website was tested for accessibility. The audits mostly follow the [WCAG Evaluation Method](https://www.w3.org/WAI/test-evaluate/conformance/wcag-em/). This method provides auditors and testers with guidelines for checking whether a success criterion has been met (and thus whether a guideline has been followed correctly). 

> 

Author: Also refer to the topc test for accessibility here wpaccessibility.org/docs/testing
